### PR TITLE
use new mmr root as commitment payload

### PR DESCRIPTION
**Pull request by andresilva**
Currently just reading from `ConsensusLog` digest.

### CODE REVIEW

File: beefy-gadget/src/lib.rs [hunk 1]
```diff
@@ -165,7 +169,7 @@ struct BeefyWorker<Block: BlockT, Id, Signature, FinalityNotifications> {
 	local_id: Id,
 	key_store: SyncCryptoStorePtr,
 	min_interval: u32,
-	rounds: Rounds<Block::Hash, NumberFor<Block>, Id, Signature>,
+	rounds: Rounds<H256, NumberFor<Block>, Id, Signature>,
```
Comment (tomusdrw) [inline beefy-gadget/src/lib.rs:172]: Could we do `type MmrRootHash = H256;`? It took me a while to figure out what that `H256` here is.

File: beefy-gadget/src/lib.rs [hunk 2]
```diff
@@ -231,13 +235,20 @@ where
...
-		debug!(target: "beefy", "🥩 Finality notification: {:?}", notification);
+		debug!(target: "beefy", "Finality notification: {:?}", notification);
 
 		if self.should_vote_on(*notification.header.number()) {
-			// TODO: this needs to be properly populated by signing an MMR root as the payload
-			// (and/or abstracting the "thing to sign") and with support for validator set changes.
+			let mmr_root = if let Some(hash) = find_mmr_root_digest::<Block, Id>(&notification.header) {
+				hash
+			} else {
+				debug!(target: "beefy", "No MMR root digest found for: {:?}", notification.header.hash());
```
Comment (tomusdrw) [inline beefy-gadget/src/lib.rs:244]: Maybe make it a warning?
```suggestion
				warn!(target: "beefy", "🥩 No MMR root digest found for: {:?}", notification.header.hash());
```

File: beefy-gadget/src/lib.rs [hunk 3]
```diff
@@ -231,13 +235,20 @@ where
 	}
 
 	fn handle_finality_notification(&mut self, notification: FinalityNotification<Block>) {
-		debug!(target: "beefy", "🥩 Finality notification: {:?}", notification);
+		debug!(target: "beefy", "Finality notification: {:?}", notification);
```
Comment (tomusdrw) [inline beefy-gadget/src/lib.rs:238]: I liked the icon, any reason for removing? :)

Comment (andresilva) [reply beefy-gadget/src/lib.rs:238]: I kept it only for "user-level" messages, i.e. info/warn.

File: primitives/src/lib.rs [hunk 4]
```diff
@@ -82,8 +85,8 @@ pub enum ConsensusLog<AuthorityId: Codec> {
 	#[codec(index = "2")]
 	OnDisabled(AuthorityIndex),
 	/// MMR root hash.
-	#[codec(index = "2")]
-	MmrRoot(H256),
+	#[codec(index = "3")]
```
Comment (tomusdrw) [inline primitives/src/lib.rs:88]: Good catch :+1:

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**